packages/dvilela/skills/memeooorr_abci/handlers.py

```diff
@@ -600,6 +665,128 @@ def _handle_get_static_file(
             index_html = file.read()
         self._send_ok_response(http_msg, http_dialogue, index_html, "text/html")
 
+    def _send_message(
+        self,
+        message: Message,
+        dialogue: Dialogue,
+        callback: Callable,
+        callback_kwargs: Optional[Dict] = None,
+    ) -> None:
+        """
+        Send a message and set up a callback for the response.
+
+        :param message: the Message to send
+        :param dialogue: the Dialogue context
+        :param callback: the callback function upon response
+        :param callback_kwargs: optional kwargs for the callback
+        """
+        self.context.outbox.put_message(message=message)
+        nonce = dialogue.dialogue_label.dialogue_reference[0]
+        self.context.state.req_to_callback[nonce] = (callback, callback_kwargs or {})
+        self.context.state.in_flight_req = True
+
+    def _handle_post_process_prompt(
+        self, http_msg: HttpMessage, http_dialogue: HttpDialogue
+    ) -> None:
+        """
+        Handle POST requests to process user prompts.
+
+        :param http_msg: the HttpMessage instance
+        :param http_dialogue: the HttpDialogue instance
+        """
+
+        request_id = http_dialogue.dialogue_label.dialogue_reference[0]
+        self.context.state.request_queue.append(request_id)
```

Comment:
I see we always append, but dont we clear the queue when the request has been served?